xcube/util/progress.py

```diff
@@ -128,6 +130,100 @@ def deactivate(self):
         _ProgressContext.instance().remove_observer(self)
 
 
+class ThreadedProgressObserver(ProgressObserver):
```

Comment:
This is not following the discussewd Delegate-Pattern, because there are no delegation made to `self.self._delegate.on_xxx()`.